src/main/java/at/hannibal2/skyhanni/features/misc/GeorgeDisplay.kt

```diff
@@ -0,0 +1,86 @@
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class GeorgeDisplay {
+
+    private val config get() = SkyHanniMod.feature.misc.pets.george
+
+    private val ROW_OFFSET: Int = 9 * 4
+    private val INDEX_OFFSET: Int = 6 - 1
+
+    private val patternGroup = RepoPattern.group("george.tamingcap")
+    private val neededPetPattern by patternGroup.pattern(
+        "needed.pet.loreline",
+        " +(?<fullThing>(?<tierColorCodes>§.)*(?<tier>(?:[uU][nN])?[cC][oO][mM][mM][oO][nN]|[rR][aA][rR][eE]|[eE][pP][iI][cC]|[lL][eE][gG][eE][nN][dD][aA][rR][yY]|[mM][yY][tT][hH][iI][cC]|[dD][iI][vV][iI][nN][eE]|(?:[vV][eE][rR][yY] )?[sS][pP][eE][cC][iI][aA][lL]|[uU][lL][tT][iI][mM][aA][tT][eE]|[sS][uU][pP][rR][eE][mM][eE]|[aA][dD][mM][iI][nN]) (?<pet>[\\S ]+))"
+    )
+
+    private val display = mutableListOf<Renderable>()
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryOpenEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!isEnabled()) return
+        if (event.inventoryName != "Offer Pets") return
+        val stack = event.inventoryItems[ROW_OFFSET + INDEX_OFFSET] ?: return
+        if (stack.cleanName() != "+1 Taming Level Cap") return
+        val updateList: MutableList<Renderable> = mutableListOf<Renderable>(Renderable.string("§d§lTaming 60 Cost:"))
+        var totalCost: Double = 0.0
+        loop@ for (line in stack.getLore()) {
+            neededPetPattern.matchMatcher(line) {
+                val origTierString = group("tier")
+                val tier = when (origTierString.uppercase()) {
+                    "COMMON" -> "0"
+                    "UNCOMMON" -> "1"
+                    "RARE" -> "2"
+                    "EPIC" -> "3"
+                    "LEGENDARY" -> "4"
+                    "MYTHIC" -> "5"
+                    else -> ""
+                }
+                val origPetString = group("pet") ?: ""
+                val pet = origPetString.uppercase().replace(" ", "_").removePrefix("FROST_")
+                val petPrice = "$pet;$tier".asInternalName().getPriceOrNull() ?: continue@loop
+                totalCost += petPrice
+                updateList.add(Renderable.clickAndHover(
+                    text = " §7- ${group("fullThing")}§7: §6${petPrice.addSeparators()} coins",
+                    tips = listOf("§aClick to run §e/ahsearch [Lvl 1] $origPetString §ato find this pet on the Auction House.", "§aNotes: §eMake sure to set the rarity filter to ${group("tierColorCodes")}$origTierString §eon your own! §cBooster Cookie required!"),
+                    onClick = {
+                        LorenzUtils.sendCommandToServer("ahsearch [Lvl 1] $origPetString")
+                    }
+                ))
+            }
+        }
+        updateList.addAll(listOf<Renderable>(Renderable.string("§dTotal cost §7(§6Lowest BIN§7): §6${totalCost.addSeparators()} coins"), Renderable.string("§c§lDisclaimer:§r§c Some pets are not available on the Auction House.")))
```

Comment:
Instead of the disclaimer, consider opening the new recipe if it isn't on the auction house.

Reply:
was gonna do that, but here's the cinch:

george's whole schtick is to request pets that can't be obtained from crafting. so i settled for a compromise instead in two parts:
```kt
private val notAHOrCraft: List<String> = listOf<String>("ENDERMAN", "BLACK_CAT", "SLUG", "SPIRIT", "GIRAFFE", "JELLYFISH", "BAL", "BABY_YETI", "RIFT_FERRET", "WISP")
```
```kt
val command: String = if (pet !in notAHOrCraft) "viewrecipe" else "wiki"
val whatCommandDoes: String = if (pet !in notAHOrCraft) "its crafting recipe" else "how to obtain it"
updateList.add(Renderable.clickAndHover(
    text = " §7- $displayPetString§7: §eNot on AH; ${if (pet !in notAHOrCraft) "try crafting it" else "find its Wiki article here"}.",
    tips = listOf("§4Click to run §e$command $pet §4to view $whatCommandDoes."),
    onClick = { LorenzUtils.sendCommandToServer("$command $pet") }
))
```
Still gotta rephrase the disclaimer though, give me a sec.

Reply:
(the idea of the list of strings in the first codeblock is to account for the very low likelihood that George requests a craftable pet)

Reply:
actually nevermind, changed the disclaimer to mention not including kat costs

also applied blanket "`/wiki PET`" treatment given george's whole schtick (see earlier)